### PR TITLE
Food Facility Registration schema 

### PR DESCRIPTION
**Pull request by brentzundel**
This PR adds a Food Facility Registration schema as required for the Ag Tech Demo 2024

### CODE REVIEW

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 1]
```diff
@@ -0,0 +1,178 @@
+$linkedData:
+  term: FFRCredential
```
Comment (mkhraisha) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:2]: Can we expand the name to be the full name not the acronym? I believe that's our current rule for all credential names

**Review by mkhraisha — CHANGES_REQUESTED** [review]

**Comment by mkhraisha** [comment]
also seems to error on CI

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml (hunk 1 above)
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:2]:
```suggestion
  term: FoodFacilityRegistraionCredential

```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
+$linkedData:
+  term: FFRCredential
+  '@id': https://w3id.org/traceability#FFRCredential
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:3]:
```suggestion
  '@id': https://w3id.org/traceability#FoodFacilityRegistrationCredential

```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        - https://w3id.org/traceability/v1
+  type:
+    type: array
+    readOnly: true
+    const:
+      - VerifiableCredential
+      - FFRCredential
+    default:
+      - VerifiableCredential
+      - FFRCredential
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:35]:
```suggestion
      - FoodFacilityRegistrationCredential
    default:
      - VerifiableCredential
      - FoodFacilityRegistrationCredential

```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    items:
+      type: string
+      enum:
+        - VerifiableCredential
+        - FFRCredential
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:40]:
```suggestion
        - FoodFacilityRegistrationCredential

```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+  credentialSchema:
+    type: object
+    properties:
+      id:
+        title: Id
+        description: The url of the schema file to be used to validate the shape of the JSON object
+        type: string
+        format: uri
+        example: https://w3id.org/traceability/openapi/components/schemas/credentials/FFRCredential.yml
+        default: https://w3id.org/traceability/openapi/components/schemas/credentials/FFRCredential.yml
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:80]:
```suggestion
        example: https://w3id.org/traceability/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml
        default: https://w3id.org/traceability/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml
```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        readOnly: true
+      type:
+        title: Type
+        description: The type of validation to be run against the defined schema.
+        const: OpenApiSpecificationValidator2022
+  credentialSubject:
+    title: FFR
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:87]:
```suggestion
    title: Food Facility Registration
```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+              - D, Facility is a retail food establishment (21 CFR 1.226(c); 1.227(b)(11))
+              - E, Facility is a non-processing fishing vessel (21 CFR 1.226(f)
+              - F, Facility is a non-bottled drinking water collection and distribution establishment (21 CFR 1.227(b)(2))
+              - H, Facility is a farm (21 CFR 1.227 and 21 CFR 1.226(b))
+              - K, Unable to determine the registration number of the manufacturer
+              - Null
+        additionalProperties: false
+    additionalProperties: false    
+    required:
+      - type
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:133]:
```suggestion
```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+      "https://w3id.org/traceability/v1"
+    ],
+    "id": "http://example.com/dd0c6f9a-5df6-40a3-bb34-863cd1fda606",
+    "credentialSchema": {
+      "type": "OpenApiSpecificationValidator2022",
+      "id": "https://w3id.org/traceability/openapi/components/schemas/credentials/FFRCredential.yml"
+    },
+    "type": [
+      "VerifiableCredential",
+      "FFRCredential"
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml]:
```suggestion
      "id": "https://w3id.org/traceability/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml"
    },
    "type": [
      "VerifiableCredential",
      "FoodFacilityRegistrationCredential"
```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+            enum:
+              - A, Facility is out of business
+              - B, Facility is a private residence (21 CFR 1.227(b)(2))
+              - C, Facility is a restaurant (21 CFR 1.226(d); 1.227(b)(10))
+              - D, Facility is a retail food establishment (21 CFR 1.226(c); 1.227(b)(11))
+              - E, Facility is a non-processing fishing vessel (21 CFR 1.226(f)
+              - F, Facility is a non-bottled drinking water collection and distribution establishment (21 CFR 1.227(b)(2))
+              - H, Facility is a farm (21 CFR 1.227 and 21 CFR 1.226(b))
+              - K, Unable to determine the registration number of the manufacturer
+              - Null
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:122]: None
```suggestion
              - None
```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+    "issuanceDate": "2022-11-01T10:58:45-04:00",
+    "credentialSubject": {
+      "type": [
+        "FFR"
+      ],
+      "id": "did:web:organization.example.com",
+      "number": "12345678901",
+      "exemption": {
+        "exempt": false,
+        "code": null
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:166]:
```suggestion
        "code": "None"
```

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+    ],
+    "issuer": {
+      "id": "did:web:qualified-organization.example.com",
+      "name": "US-FDA"
+    },
+    "issuanceDate": "2022-11-01T10:58:45-04:00",
+    "credentialSubject": {
+      "type": [
+        "FFR"
+      ],
```
Comment (brentzundel) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:161]:
```suggestion
```

**Review by mkhraisha — APPROVED** [review]

File: docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+        type: string
+    additionalProperties: false
+    required:
+      - id
+  credentialSchema:
+    type: object
+    properties:
+      id:
+        title: Id
+        description: The url of the schema file to be used to validate the shape of the JSON object
```
Comment (TallTed) [inline docs/openapi/components/schemas/credentials/FoodFacilityRegistrationCredential.yml:80]:
```suggestion
        description: The URL of the schema file to be used to validate the shape of the JSON object

```

**Review by TallTed — CHANGES_REQUESTED** [review]
Tiny, human-facing

**Review by nissimsan — APPROVED** [review]

**Pull request merged**